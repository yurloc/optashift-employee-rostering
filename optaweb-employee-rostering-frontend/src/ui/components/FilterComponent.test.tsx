/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { shallow } from 'enzyme';
import * as React from 'react';
import { Button, TextInput } from '@patternfly/react-core';
import { useTranslation, WithTranslation } from 'react-i18next';
import { FilterProps, FilterComponent } from './FilterComponent';

describe('Filter component', () => {
  it('should render correctly with no text', () => {
    const myProps = { ...props, filterText: '' };
    const filter = shallow(<FilterComponent {...myProps} />);
    expect(filter).toMatchSnapshot();
  });

  it('should render correctly with text', () => {
    const filter = shallow(<FilterComponent {...props} />);
    expect(filter).toMatchSnapshot();
  });

  it('should call filter and onChange on updateFilter', () => {
    const filter = shallow(<FilterComponent {...props} />);
    filter.find(TextInput).simulate('change', 'New Filter');
    expect(props.onChange).toBeCalled();
    expect(props.onChange).toBeCalledWith('New Filter');
  });

  it('should call updateFilter with empty text when button is clicked', () => {
    const filter = shallow(<FilterComponent {...props} />);

    filter.find(Button).simulate('click');
    expect(props.onChange).toBeCalled();
    expect(props.onChange).toBeCalledWith('');
  });
});

const props: FilterProps & WithTranslation = {
  ...useTranslation('FilterComponent'),
  tReady: true,
  filterText: 'Test',
  onChange: jest.fn(),
};
