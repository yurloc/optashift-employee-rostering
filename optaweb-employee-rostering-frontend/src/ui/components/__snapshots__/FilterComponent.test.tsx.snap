// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component should render correctly with no text 1`] = `
<div
  className="search-icons"
>
  <TextInput
    aria-label="Search"
    className=""
    isDisabled={false}
    isReadOnly={false}
    isRequired={false}
    isValid={true}
    onChange={[MockFunction]}
    placeholder="Trans(i18nKey=search)"
    type="text"
    value=""
  />
</div>
`;

exports[`Filter component should render correctly with text 1`] = `
<div
  className="search-icons"
>
  <TextInput
    aria-label="Search"
    className=""
    isDisabled={false}
    isReadOnly={false}
    isRequired={false}
    isValid={true}
    onChange={[MockFunction]}
    placeholder="Trans(i18nKey=search)"
    type="text"
    value="Test"
  />
  <Button
    onClick={[Function]}
    variant="plain"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      height="1em"
      role="img"
      style={
        Object {
          "verticalAlign": "-0.125em",
        }
      }
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
        transform=""
      />
    </svg>
  </Button>
</div>
`;
